src/main/java/freiman/earthquake/EarthquakeFrame.java

```diff
@@ -0,0 +1,116 @@
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+
+public class EarthquakeFrame extends JFrame {
+
+    private JList<String> jlist = new JList<>();
+    private ListSelectionModel listSelectionModel;
+    private EarthquakeService service = new EarthquakeServiceFactory().getService();
+
+    private Double longitude;
+    private Double latitude;
+    private FeatureCollection featureCollection;
+
+    public EarthquakeFrame() {
+
+        setTitle("EarthquakeFrame");
+        setSize(300, 600);
+        setDefaultCloseOperation(EXIT_ON_CLOSE);
+
+        setLayout(new BorderLayout());
+
+
+        // Create the radio buttons and add to a group
+        JRadioButton lastHour = new JRadioButton("One Hour");
+        lastHour.setActionCommand("lastHour");
+        JRadioButton lastMonth = new JRadioButton("30 Days");
+        lastMonth.setActionCommand("lastMonth");
+
+        ButtonGroup buttonGroup = new ButtonGroup();
+        buttonGroup.add(lastHour);
+        buttonGroup.add(lastMonth);
+
+        JPanel radioButtonPanel = new JPanel();
+        radioButtonPanel.add(lastHour);
+        radioButtonPanel.add(lastMonth);
+
+        lastHour.addActionListener(e -> radioActionHandler(e));
+        lastMonth.addActionListener(e -> radioActionHandler(e));
+
+        listSelectionModel = jlist.getSelectionModel();
+        listSelectionModel.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+        listSelectionModel.addListSelectionListener(
+                new ListSelectionListener() {
+                    public void valueChanged(ListSelectionEvent e) {
+                        int selIndex = jlist.getSelectedIndex();
+
+                        if (selIndex != -1) {
+                            Feature feature = featureCollection.features[selIndex];
+                            longitude = feature.geometry.coordinates[0];
+                            latitude = feature.geometry.coordinates[1];
+
+                            String url = "http://www.google.com/maps/place/" + latitude + "," + longitude;
+                            if (Desktop.isDesktopSupported()
+                                    && Desktop.getDesktop().isSupported(Desktop.Action.BROWSE)) {
+                                try {
+                                    Desktop.getDesktop().browse(new URI(url));
+                                } catch (IOException | URISyntaxException ex) {
+                                    throw new RuntimeException(ex);
+                                }
+                            }
+                        }
+                    }
+                });
+
+        // Add to the frame
+        add(radioButtonPanel, BorderLayout.NORTH);
+        add(jlist, BorderLayout.CENTER);
+    }
+
+    private void radioActionHandler(ActionEvent e) {
+        String actionCommand = e.getActionCommand();
+        if ("lastHour".equals(actionCommand)) {
```

Comment:
`e.getSource() == lastHour`

You can do this if you make `lastHour` a member variable.

However it might be better to just have two "actionHandler" methods, one for `oneHour` and the other for `lastMonth` so that you don't need the if statement at all.